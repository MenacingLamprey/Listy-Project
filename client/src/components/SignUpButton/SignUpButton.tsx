import { useAuth0 } from '@auth0/auth0-react'

const SignUpButton = () => {

  const { loginWithRedirect, isAuthenticated } = useAuth0();
    return (

      <div>
        { !isAuthenticated ? (
          <button className='signup-button' onClick={() => loginWithRedirect({
            screen_hint: 'signup',
          })}>
            Sign Up
          </button>
        ) : (
          null
        )}
      </div>

      //MV todo: clean up this commented code when we confirm ternary refactor works
      // !isAuthenticated && (
      //   <button className='signup-button' onClick={() => loginWithRedirect({
      //     screen_hint: 'signup',
      //   })}>
      //       Sign Up
      //   </button>
      //   )
    )
}

export default SignUpButton;