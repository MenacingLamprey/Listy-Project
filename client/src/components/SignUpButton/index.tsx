import React, { FunctionComponent } from 'react'
// import { useAuth0 } from '@auth0/auth0-react'

interface IProps {
  isAuthenticated: boolean,
  loginWithRedirect: Function,
  loginWithPopup: Function,
}

const SignUpButton: FunctionComponent<IProps> = ({isAuthenticated, loginWithRedirect, loginWithPopup}) => {

    return (

      <div className='container'>
        { !isAuthenticated ? (
          <button className='signup-button' onClick={() => loginWithRedirect({
            screen_hint: 'signup',
          })}>
            Sign Up
          </button>
        ) : (
          null
        )}
      </div>
    )
}

export default SignUpButton;