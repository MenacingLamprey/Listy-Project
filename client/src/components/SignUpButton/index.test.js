import { screen, render } from "@testing-library/react";
import SignUpButton from './index';
import '@testing-library/jest-dom';

describe('Sign-Up Button', () => {
  test('Should render a button', () => {
    const { container } = render(<SignUpButton />);
    expect(container).toMatchSnapshot();
  })

  test('Should contain label "Sign Up"', () => {
    render(<SignUpButton />);
    screen.getByText(/Sign Up/);
  })

  test('Should not render if authenticated', () => {
    render(<SignUpButton isAuthenticated={true} />);
    // eslint-disable-next-line testing-library/no-container
    expect(screen.queryByText(/Sign Up/)).not.toBeInTheDocument();
  })
})