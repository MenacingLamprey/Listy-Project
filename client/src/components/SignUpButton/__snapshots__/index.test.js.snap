// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign-Up Button Should render a button 1`] = `
<div>
  <div
    class="container"
  >
    <button
      class="signup-button"
    >
      Sign Up
    </button>
  </div>
</div>
`;
